cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -200,4 +201,154 @@ public String toString() {
       return sb.toString();
     }
   }
+
+  /**
+   * Represents ID of a Notification feed.
+   */
+  public static class NotificationFeed {
+
+    /**
+     * Id is composed of: [namespaceId].[category].[name]
+     */
+    private final String id;
+
+    private final Namespace namespace;
+    private final String category;
+    private final String name;
+
+    private final String description;
+
+    /**
+     * {@link NotificationFeed} object from an id in the form of "namespace.category.name".
+     *
+     * @param id id of the notification feed to build
+     * @return a {@link NotificationFeed} object which id is the same as {@code id}
+     * @throws IllegalArgumentException when the id doesn't match a valid feed id
+     */
+    public static NotificationFeed fromId(String id) {
+      String[] idParts = id.split("\\.");
+      if (idParts.length != 3) {
+        throw new IllegalArgumentException(String.format("Id %s is not a valid feed id.", id));
+      }
+      return new NotificationFeed(idParts[0], idParts[1], idParts[2], "");
+    }
+
+    private NotificationFeed(String namespace, String category, String name, String description) {
+      Preconditions.checkArgument(namespace != null && !namespace.isEmpty(),
+                                  "Namespace value cannot be null or empty.");
+      Preconditions.checkArgument(category != null && !category.isEmpty(), "Category value cannot be null or empty.");
+      Preconditions.checkArgument(name != null && !name.isEmpty(), "Name value cannot be null or empty.");
+      Preconditions.checkArgument(isId(namespace) && isId(category) && isId(name),
+                                  "Namespace, category or name has a wrong format.");
+
+      this.namespace = Namespace.from(namespace);
+      this.category = category;
+      this.name = name;
+      this.id = String.format("%s.%s.%s", namespace, category, name);
+      this.description = description;
+    }
+
+    private boolean isId(String name) {
+      return CharMatcher.inRange('A', 'Z')
+        .or(CharMatcher.inRange('a', 'z'))
+        .or(CharMatcher.is('-'))
+        .or(CharMatcher.is('_'))
+        .or(CharMatcher.inRange('0', '9')).matchesAllOf(name);
+    }
+
+    public String getCategory() {
+      return category;
+    }
+
+    public String getId() {
+      return id;
+    }
+
+    public String getNamespaceId() {
+      return namespace.getId();
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getDescription() {
+      return description;
+    }
+
+    /**
+     * Builder used to build {@link NotificationFeed}.
+     */
+    public static final class Builder {
```

Comment:
Could ``Id.NotificationFeed`` grow to contain a lot more fields? Just thinking that its construction (via ``Builder``) is not consistent with the other ``Id`` classes (which expose the constructor directly).